processor/coralogixprocessor/README.md

```diff
@@ -0,0 +1,85 @@
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@crobert-1](https://www.github.com/crobert-1), [@galrose](https://www.github.com/galrose) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+<!-- end autogenerated section -->
+
+## Description
+
+The Coralogix processor adds attributes to spans that enable features in Coralogix.
+
+## Features
+
+### DB Statement Blueprints
+
+This feature enables the processor to create blueprints from SQL queries, this means replacing any variables with `?`.
+The blueprint is also hashed to be able to be used with spanmetrics connector.
+Long queries can be an issue when being stored in certain metric stores.
+Blueprints alleviate this problem by using the hash as the identifying dimension on the metric, which enables
+users to query metrics by blueprints.
+
+The added attributes are `db.statement.blueprint` and `db.statement.blueprint.id`.
+
+* `db.statement.blueprint` contains the blueprinted version of the statement, we require them to be sent to coralogix to
+  display your blueprinted statement
+* `db.statement.blueprint.id` contains a hash of the statement, this way we can add it as a dimension in the spanmetrics
+  connector and use it to query your blueprints.
+* `sampling.priority` if enabled contains the value 100 for new blueprints, further explanation below.
+
+#### Sampling
+
+If sampling is enabled then it stores the found blueprints in an in-memory cache to be able to send only new blueprints
+that haven't been seen yet.
+This only adds an attribute to the span named `sampling.priority`, if the blueprint is new then the sampling priority
+will be `100`.
+
+Using this key it's possible to use either
+the [Tail Sampler](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor)
+or
+the [Probabilistic Sampler](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor)
+to only send new blueprints to coralogix.
+If sampling is not enabled it won't cache anything and the `sampling.priorty` attribute won't be added.
+
+The cache is limited by the `max_cache_size_mib` configuration, if the cache is full it will remove the oldest entries
+to make space for new ones.
+The cache stores hashes of the queries, each hash is 8 bytes, so the number of maximum cache entries is calculated
+by `max_cache_size_mib * 1024 * 1024 / 8`.
+
+## Config
+
+* `db_statment_blueprints`
+    * `sampling`: If enabled, adds the attribute `sampling.priority` with a value of `100` to spans with new blueprints.
+      Refer to the [Sampling section](#sampling) for more information.
+        * `max_cache_size_mib` (default: `1024`) The size of the cache in mebibytes to store seen blueprints hashes.
```

Comment:
Do we want to add a bool `enabled` config option here to show if caching is enabled or not? Without it

Reply:
The question that I'm getting at is with the current config, how would a user disable sampling? It's not obvious from current configuration options and the README.

Reply:
Im not sure what the best way to describe it in the README but just not adding the key means not enabling it, similar to other processors, do you have a good idea how I should explain it?

Reply:
I added a comment about it in this [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33458/commits/d68c182a69d167fc0fbccd2efb86c6a07c7fc8d6), do you think this is enough?

Reply:
A component's configuration has no way of knowing if a key is included in the configuration, as far as I know. The collector uses [mapstructure](https://github.com/go-viper/mapstructure) to unmarshal YAML files into configuration options. If there's no value for a key (`sampling` in this case), there's nothing to unmarshal. You'd need a value to be stored to know if a configuration option was specified or not.

If I've missed something please let me know. (You mention other processors do this, so if my explanation doesn't make sense, feel free to give an example.)

Reply:
[2dd6c34](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33458/commits/2dd6c3417678af6d05b2803b5e69192f8535851c)

added the enabled key